tools/scripts/components-update-sdc.js

```
@@ -242,54 +242,169 @@ function readYamlFile(yamlPath) {
+      // Basic slot metadata
+      result.slots[slotName] = {
+        title: slotData.title || slotName,
+        description: slotData.description || '',
+      };
+      
+      // If the slot has items (like for repeating content)
+      if (slotData.items) {
+        result.slots[slotName].items = slotData.items;
+      }
+    });
+  }
+
+  // Handle blocks documentation - check if the twig file exists for this component
+  const dir = path.dirname(yamlPath);
+  const componentName = path.basename(yamlPath, '.component.yml');
+  const twigPath = path.join(dir, `${componentName}.twig`);
+  
+  // First check if there's an existing docblock with blocks documentation
+  if (fs.existsSync(twigPath)) {
+    const twigContent = fs.readFileSync(twigPath, 'utf8');
+    const docblockMatch = twigContent.match(/{#\s*\/\*\*[\s\S]*?\*\/\s*#}/);
+    
+    if (docblockMatch && docblockMatch[0].includes(' * Blocks:')) {
+      // If there's existing blocks documentation, parse it
+      const blockLines = docblockMatch[0].split('\n')
+        .filter(line => line.includes(' * - ') && line.includes(': [block]'))
+        .map(line => {
+          // Extract block name and description from existing docblock
+          const matches = line.match(/ \* - ([a-zA-Z0-9_]+): \[block\] (.*)/);
+          if (matches && matches.length >= 3) {
+            return { name: matches[1], description: matches[2] };
+          }
+          return null;
+        })
+        .filter(Boolean);
+        
+      // Add parsed blocks to result
+      blockLines.forEach(block => {
+        result.blocks[block.name] = {
+          description: block.description,
+        };
+      });
+    } else {
+      // If no blocks documentation exists, extract blocks from twig file
+      const blocks = extractTwigBlocks(twigPath);
+      
+      // For each block, try to find a corresponding slot with a similar name
+      Object.keys(blocks).forEach(blockName => {
```

And here is that logic below of what to do when its missing in the component - might be better to throw an error here or update `component.yml`